src/CommandBuilder/CommandBuilderInterface.php

```diff
@@ -0,0 +1,18 @@
+<?php
+
+namespace Castor\CommandBuilder;
+
+use Castor\Context;
+
+interface CommandBuilderInterface
+{
+    /**
+     * @return string|array<string|\Stringable|int>
+     */
+    public function getCommand(): array|string;
+
+    /**
+     * @return (callable(Context): Context)[]
+     */
+    public function getContextModifiers(): array;
```

Comment:
To simplify the usage, I would do:

```suggestion
    public function getContext(Context $context): Context;
```

Then, it's up to the Builder to update the context